spec/message.abnf

```diff
@@ -42,15 +42,15 @@ function = ":" name
 markup-start = "+" name
 markup-end = "-" name
 
-name    = name-start *name-char ; matches XML https://www.w3.org/TR/xml/#NT-Name
-nmtoken = 1*name-char           ; matches XML https://www.w3.org/TR/xml/#NT-Nmtokens
+name    = name-start *name-char ; based on https://www.w3.org/TR/xml/#NT-Name, but cannot start with U+003A COLON ":"
+nmtoken = 1*name-char           ; equal to https://www.w3.org/TR/xml/#NT-Nmtoken
```

Comment:
Is the colon in `{:functionName}` syntax sufficiently valuable to justify deviation from **exact** match of the XML productions? If so, I question whether alignment with XML is actually well-motivated, and if not then perhaps it would make more sense to use characters excluded from Nmtoken—``` !"#$%&'()*+,/;<=>?@[\]^`{|}~ ```.

Note also that if leading `:` is prohibited because of [`function`](https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md#names), then leading `-` should be prohibited for the same reason.
```suggestion
name    = name-start *name-char ; based on https://www.w3.org/TR/xml/#NT-Name, but cannot
                                ; start with U+003A COLON ":" or U+002D HYPHEN-MINUS "-"
nmtoken = 1*name-char           ; equal to https://www.w3.org/TR/xml/#NT-Nmtoken
```